python/tests/integration/connectors/memory/test_postgres.py

```diff
@@ -0,0 +1,250 @@
+
+    await memory.create_collection_async("test_collection", 2)
+    # await memory.create_collection_async("test_collection-b")
+    # await memory.create_collection_async("test_collection-c")
+    result = await memory.get_collections_async()
+    assert len(result) > 1
+
+
+@pytest.mark.asyncio
+async def test_delete_collection_async(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    await memory.delete_collection_async("test_collection")
+    result = await memory.get_collections_async()
+    assert "test_collection" not in result
+
+
+@pytest.mark.asyncio
+async def test_does_collection_exist_async(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    result = await memory.does_collection_exist_async("test_collection")
+    assert result is True
+
+
+@pytest.mark.asyncio
+async def test_upsert_async_and_get_async(get_postgres_config, memory_record1):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    await memory.upsert_async("test_collection", memory_record1)
+    result = await memory.get_async(
+        "test_collection", memory_record1._id, with_embedding=True
+    )
+    assert result is not None
+    assert result._id == memory_record1._id
+    assert result._text == memory_record1._text
+    assert result.embedding is not None
+
+
+@pytest.mark.asyncio
+async def test_upsert_batch_async_and_get_batch_async(
+    get_postgres_config, memory_record1, memory_record2
+):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    await memory.upsert_batch_async("test_collection", [memory_record1, memory_record2])
+
+    results = await memory.get_batch_async(
+        "test_collection",
+        [memory_record1._id, memory_record2._id],
+        with_embeddings=True,
+    )
+
+    assert len(results) >= 2
+    assert results[0]._id in [memory_record1._id, memory_record2._id]
+    assert results[1]._id in [memory_record1._id, memory_record2._id]
+
+
+@pytest.mark.asyncio
+async def test_remove_async(get_postgres_config, memory_record1):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    await memory.upsert_async("test_collection", memory_record1)
+    await memory.remove_async("test_collection", memory_record1._id)
+    with pytest.raises(KeyError):
+        _ = await memory.get_async(
+            "test_collection", memory_record1._id, with_embedding=True
+        )
```

Comment:
Same as for collections, I would add extra `get` operation just to double-check that deletion actually happened:
```suggestion
    await memory.create_collection_async("test_collection", 2)
    await memory.upsert_async("test_collection", memory_record1)
    
    result = await memory.get_async(
        "test_collection", memory_record1._id, with_embedding=True
    )
    assert result is not None
    
    await memory.remove_async("test_collection", memory_record1._id)
    with pytest.raises(KeyError):
        _ = await memory.get_async(
            "test_collection", memory_record1._id, with_embedding=True
        )
```